src/lsptoolshost/copilot/contextProviders.ts

```diff
@@ -77,26 +88,35 @@ export function registerCopilotContextProviders(
     devkit.activate().then(async (devKitExports) => {
         try {
             // Check if the Copilot Language Server extension is installed and has the correct capabilities
-            let hasCapabilities = false;
+            // 0 means not support, 1 means old version, 2 means new version
+            let hasCapabilities = 0;
             const copilotServerExtensionfolder = devKitExports.components[copilotLanguageServerExtensionComponentName];
             if (copilotServerExtensionfolder) {
                 const capabilitiesFilePath = path.join(
                     copilotServerExtensionfolder,
                     copilotLanguageServerExtensionCapabilitiesFileName
                 );
                 const capabilitiesContent = await readJsonSync(capabilitiesFilePath);
-                if (
-                    capabilitiesContent?.capabilities?.find(
-                        (capability: any) =>
-                            capability?.method === resolveContextMethodName &&
-                            capability?.version === resolveContextMethodSupportedVersion
-                    )
-                ) {
-                    hasCapabilities = true;
+                for (const capability of capabilitiesContent?.capabilities ?? []) {
+                    if (
+                        capability.method === oldResolveContextMethodName &&
+                        capability.version === oldresolveContextMethodSupportedVersion
+                    ) {
+                        hasCapabilities = 1;
+                        channel.debug(`supported 'roslyn/resolveContext' method found in capabilities.json`);
+                        break;
+                    } else if (
+                        capability.method === newResolveContextMethodName &&
+                        capability.version === newResolveContextMethodSupportedVersion
+                    ) {
+                        hasCapabilities = 2;
+                        channel.debug(`supported 'roslyn/resolveContext@2' method found in capabilities.json`);
```

Comment:
need to make a note here - this PR requires a Roslyn version bump with the handler present in it (otherwise this request may fail).